var/spack/repos/builtin/packages/dyninst/package.py

```diff
@@ -41,7 +42,8 @@ class Dyninst(CMakePackage):
 
     # Dyninst uses elf@1 (elfutils) starting with 9.3.0, and used
     # elf@0 (libelf) before that.
-    depends_on('elf@1', type='link', when='@9.3.0:')
+    depends_on('elf@0.173:', type='link', when='@10.2.0:')
```

Comment:
Should this be elfutils? Since you need a specific version of the library, it might be better to be explicit.

Reply:
It would be ok with me to change all the `elf@0` to libelf and `elf@1`
to elfutils.  I originally wrote them as `elf@1` because I though that
was the more spack-ish style.  (`elf` is a meta-package, like mpi.)
But this may be a case where it's more awkward than it's worth.

Reply:
I tried this first, but had to revert it (c018267) because it tickled a bug in the concretization process. I was planning to file a bug on that separately.